pages/studies.py

```diff
@@ -83,12 +82,13 @@ def study_download_zip(studyId):
 
         csv_data.append((csv_name, csv_bytes))
 
-    study = study_dfs.get_general_info(studyId, g.db_session)
+    study = g.db_session.scalars(sql.select(Study).where(Study.studyId == studyId)).one()
```

Comment:
Funny question: why do we need the `one()` in the end of this line ?

Reply:
The `sql.select(...).where(...)` portion returns a query object that can be converted to an SQL string. The `.scalars(...)` function takes that and sends it to the database and returns a "result" object. After that, to fetch the results, we need to call another function that takes `.all()` results, or `.one()` result, or maybe `.one_or_none()`. I call `.one()` here to raise an error if there's more than one result or less than one.

This study id is actually the primary key in the database, but I didn't mark it as such in the ORM, otherwise I could have run `g.db_session.get(Study, studyId)`. I was wondering if that should be the primary key or the unique id. I'll do another pass later to stabilize the design of these tables.

Reply:
I guess `studyId`, e.g. `SMGDB00000002` could be the `primary` but I have to go through the whole structure and I have not done that yet.. :unamused: